src/sentry/services/hybrid_cloud/integration/impl.py

```diff
@@ -429,16 +429,28 @@ def delete_integration(self, *, integration_id: int) -> None:
     def get_integration_external_project(
         self, *, organization_id: int, integration_id: int, external_id: str
     ) -> RpcIntegrationExternalProject | None:
-        external_project = IntegrationExternalProject.objects.filter(
+        external_projects = self.get_integration_external_projects(
+            organization_id=organization_id,
+            integration_id=integration_id,
             external_id=external_id,
-            organization_integration_id__in=OrganizationIntegration.objects.filter(
-                organization_id=organization_id,
-                integration_id=integration_id,
-            ),
+        )
+        return external_projects[0] if len(external_projects) > 0 else None
+
+    def get_integration_external_projects(
+        self, *, organization_id: int, integration_id: int, external_id: str | None = None
+    ) -> List[RpcIntegrationExternalProject]:
+        oi = OrganizationIntegration.objects.filter(
+            organization_id=organization_id,
+            integration_id=integration_id,
         ).first()
-        if external_project is None:
-            return None
-        return serialize_integration_external_project(external_project)
+        if not oi:
+            return []
```

Comment:
Another nit for clarity:
```suggestion
        try:
            oi = OrganizationIntegration.objects.get(
                organization_id=organization_id, integration_id=integration_id
            )
        except OrganizationIntegration.DoesNotExist:
            return []
```

This should be equivalent because the `filter` condition guarantees uniqueness. It's just a preference of mine that `first` should be used only when more than one result is possible and order is significant.

Reply:
:+1: to biasing towards get() when we only expect a single row.